plugins/guests/redhat/cap/change_host_name.rb

```diff
@@ -29,7 +29,13 @@ def self.change_host_name(machine, name)
               }
 
               # Restart network
-              service network restart
+              if test -f /etc/init.d/network; then
```

Comment:
i still think we need to check NM first here. It's possible a system has both legacy network plus NetworkManager installed. Checking `systemctl -q is-active NetworkManager.service` first would be best. If it's not active and network scripts exist then try that. If network scripts don't exist then error out.

Reply:
I am fine with that but older vagrant images like CentOS will bomb out. I will add in a check to ensure that systemctl is present when doing that check first.

```
# systemctl
-bash: systemctl: command not found
# cat /etc/redhat-release 
CentOS release 6.10 (Final)```

Reply:
> I am fine with that but older vagrant images like CentOS will bomb out.

good point. so we can say `if -e /usr/bin/systemctl && systemctl -q is-active NetworkManager.service` or something along those lines